cmd/rebasehelpers/util/git.go

```diff
@@ -0,0 +1,97 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Commit struct {
+	Sha     string
+	Summary string
+	Files   []File
+}
+
+func (c Commit) DeclaresUpstreamChange() bool {
+	return strings.Contains(strings.ToLower(c.Summary), "upstream")
+}
+
+func (c Commit) DeclaresUpstreamRevert() bool {
+	return c.DeclaresUpstreamChange() && strings.Contains(strings.ToLower(c.Summary), "revert")
+}
+
+func (c Commit) HasGodepsChanges() bool {
+	for _, file := range c.Files {
+		if file.HasGodepsChanges() {
+			return true
+		}
+	}
+	return false
+}
+
+type File string
+
+func (f File) HasGodepsChanges() bool {
+	return strings.Contains(string(f), "Godeps")
```

Comment:
We're talking about the paths represented in the git tree, so HasPrefix should work.